packages/andromeda-finance/src/weighted_splitter.rs

```diff
@@ -0,0 +1,128 @@
+    pub recipients: Vec<AddressWeight>,
+    /// Whether or not the contract is currently locked. This restricts updating any config related fields.
+    pub locked: bool,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    /// The vector of recipients for the contract. Anytime a `Send` execute message is
+    /// sent the amount sent will be divided amongst these recipients depending on their assigned weight.
+    pub recipients: Vec<AddressWeight>,
+    pub modules: Option<Vec<Module>>,
+}
+
+impl InstantiateMsg {
+    pub fn validate(&self) -> Result<bool, ContractError> {
+        validate_recipient_list(self.recipients.clone())?;
+        Ok(true)
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    /// Update the recipients list. Only executable by the contract owner when the contract is not locked.
+    UpdateRecipients {
+        recipients: Vec<AddressWeight>,
+    },
+    /// Add a single recipient to the recipient list. Only executable by the contract owner when the contract is not locked.
+    AddRecipient {
+        recipient: AddressWeight,
+    },
+    /// Remove a single recipient from the recipient list. Only executable by the contract owner when the contract is not locked.
+    RemoveRecipient {
+        recipient: Recipient,
+    },
+    /// Used to lock/unlock the contract allowing the config to be updated.
+    UpdateLock {
+        lock: bool,
+    },
+    /// Divides any attached funds to the message amongst the recipients list.
+    Send {},
+    AndrReceive(AndromedaMsg),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct MigrateMsg {}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    AndrQuery(AndromedaQuery),
+    /// The current config of the Splitter contract
+    GetSplitterConfig {},
+    /// Gets user's allocated weight
+    GetUserWeight {
+        user: Recipient,
+    },
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema)]
+pub struct GetSplitterConfigResponse {
+    pub config: Splitter,
+}
+/// In addition to returning a specific recipient's weight, this function also returns the total weight of all recipients.
+/// This serves to put the user's weight into perspective.
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema)]
+pub struct GetUserWeightResponse {
+    pub weight: Uint128,
+    pub total_weight: Uint128,
+}
+
+/// Ensures that a given list of recipients for a `weighted-splitter` contract is valid:
+///
+/// * Must include at least one recipient
+pub fn validate_recipient_list(recipients: Vec<AddressWeight>) -> Result<bool, ContractError> {
```

Comment:
Change parameter to `&[AddressWeight]` since this function does not need ownership of the vector. I'm not 100% sure that this function is even needed since it is just a simple `require` call.